libs/core/src/lib/split-button/split-button.component.ts

```diff
@@ -79,9 +85,63 @@ export class SplitButtonComponent {
     @ContentChild(MenuComponent)
     menu: MenuComponent;
 
+    /** @hidden */
+    private _firstElement: MenuItemComponent;
+
+    get firstElement(): MenuItemComponent {
+        return this._firstElement;
+    }
+
+    set firstElement(item: MenuItemComponent) {
+        this._firstElement = item;
+    }
+
+    /** An RxJS Subject that will kill the data stream upon component’s destruction (for unsubscribing)  */
+    private readonly _onDestroy$: Subject<void> = new Subject<void>();
+
+    constructor(private _changeDetectorRef: ChangeDetectorRef) { }
+
     /** @hidden Emits event when main button is clicked */
     onMainButtonClick(event: MouseEvent): void {
-        this.primaryButtonClicked.emit(event);
+        this.mainButtonClicked();
         event.stopPropagation();
     }
+
+    ngAfterContentChecked(): void {
+        this.selectedSubscribed();
+        this.getFirstMenuItem();
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._onDestroy$.next();
+        this._onDestroy$.complete();
+    }
+
+    /** @hidden */
+    private mainButtonClicked(): void {
+        if (this.mainActionTitle) {
+            this.primaryButtonClicked.emit(event);
```

Comment:
where is "event" coming from here?

Reply:
it was from onMainButtonClicked i did this for it to be more modular but i changed it back